types.validateValue: unsupported type: types.Array

Getting the error `types.validateValue: unsupported type: types.Array` when trying to connect with `mongo` to the `ferretdb`.

Steps to reproduce:

1. Start FerretDB:

```shell
➜ make run
gotip generate -x ./internal/util/version
./version.sh
+ git describe --tags --dirty
gotip test -c -o=bin/ferretdb-testcover -trimpath -tags=testcover -race -coverpkg=./... ./cmd/ferretdb
bin/ferretdb-testcover -test.coverprofile=cover.txt -mode=diff-normal -listen-addr=:27017
2021-12-26T19:20:57.294+0300	INFO	ferretdb/main.go:61	Starting FerretDB v0.0.4-53-g22a1644-dirty...	{"commit": "22a1644e7d6bf85b24424fdf67ec502f058e9c78", "dirty": true}
2021-12-26T19:20:57.295+0300	INFO	debug	debug/debug.go:36	Starting debug server on http://127.0.0.1:8088/
2021-12-26T19:20:57.297+0300	INFO	pgconn.Pool	v4@v4.14.1/conn.go:355	Dialing PostgreSQL server	{"host": "127.0.0.1"}
2021-12-26T19:20:57.305+0300	DEBUG	pgconn.Pool	pg/pool.go:155	PostgreSQL setting	{"name": "client_encoding", "setting": "UTF8"}
2021-12-26T19:20:57.306+0300	DEBUG	pgconn.Pool	pg/pool.go:155	PostgreSQL setting	{"name": "lc_collate", "setting": "en_US.utf8"}
2021-12-26T19:20:57.306+0300	DEBUG	pgconn.Pool	pg/pool.go:155	PostgreSQL setting	{"name": "lc_ctype", "setting": "en_US.utf8"}
2021-12-26T19:20:57.307+0300	DEBUG	pgconn.Pool	pg/pool.go:155	PostgreSQL setting	{"name": "server_encoding", "setting": "UTF8"}
2021-12-26T19:20:57.308+0300	INFO	pgconn.Pool	v4@v4.14.1/conn.go:355	Query	{"sql": "SHOW ALL", "args": [], "time": "4.747591ms", "rowCount": 345, "pid": 781}
2021-12-26T19:20:57.308+0300	INFO	listener	clientconn/listener.go:66	Listening on :27017 ...

```

2. Start `mongo` shell in the second bash session:

```shell
➜ mongo --norc --port 27017
MongoDB shell version v5.0.4
connecting to: mongodb://127.0.0.1:27017/?compressors=disabled&gssapiServiceName=mongodb
WARNING: No implicit session: Logical Sessions are only supported on server versions 3.6 and greater.
Implicit session: dummy session
MongoDB server version: 5.0.42
================
Warning: the "mongo" shell has been superseded by "mongosh",
which delivers improved usability and compatibility.The "mongo" shell has been deprecated and will be removed in
an upcoming release.
For installation instructions, see
https://docs.mongodb.com/mongodb-shell/install/
================
uncaught exception: Error: error doing query: failed: network error while attempting to run command 'getLog' on host '127.0.0.1:27017'  :
DB.prototype.runCommand@src/mongo/shell/db.js:177:19
DB.prototype._runCommandWithoutApiStrict@src/mongo/shell/db.js:215:12
shellHelper.show@src/mongo/shell/utils.js:1042:17
shellHelper@src/mongo/shell/utils.js:838:15
@(shellwarnings):1:1
uncaught exception: Error: error doing query: failed: socket exception [CONNECT_ERROR] server [network error while attempting to run command 'isMaster' on host '127.0.0.1:27017' ] :
DB.prototype.runCommand@src/mongo/shell/db.js:177:19
DB.prototype.adminCommand@src/mongo/shell/db.js:195:12
shellHelper.show@src/mongo/shell/utils.js:1130:35
shellHelper@src/mongo/shell/utils.js:838:15
@(freeMonitoring):1:1
Error: socket exception [CONNECT_ERROR] server [couldn't connect to server 127.0.0.1:27017, connection attempt failed: SocketException: Error connecting to 127.0.0.1:27017 :: caused by :: Connection refused] :
runClientFunctionWithRetries@src/mongo/shell/session.js:371:27
runCommand@src/mongo/shell/session.js:466:25
DB.prototype._runCommandImpl@src/mongo/shell/db.js:155:12
DB.prototype.runCommand@src/mongo/shell/db.js:170:16
DB.prototype._helloOrLegacyHello@src/mongo/shell/db.js:205:12
shellHelper.show@src/mongo/shell/utils.js:1097:23
shellHelper@src/mongo/shell/utils.js:838:15
@(automationnotices):1:1
> 
```

3. Check errors in the first session:

```shell
2021-12-26T19:22:04.430+0300	INFO	pgconn.Pool	v4@v4.14.1/conn.go:355	Query	{"sql": "SHOW server_version", "args": [], "time": "1.197863ms", "rowCount": 1, "pid": 781}
2021-12-26T19:22:04.431+0300	DPANIC	// 127.0.0.1:52338 -> 127.0.0.1:27017 	clientconn/conn.go:118	types.MakeDocument: types.Document.validate: types.validateValue: unsupported type: types.Array ({[{"c":"STORAGE","ctx":"initandlisten","id":42000,"msg":"Powered by 🥭 FerretDB v0.0.4-53-g22a1644-dirty and PostgreSQL 14.1.","s":"I","t":{"$date":"2021-12-26T16:22:04.43Z"},"tags":["startupWarnings"]} {"c":"STORAGE","ctx":"initandlisten","id":42000,"msg":"Please star us on GitHub: https://github.com/FerretDB/FerretDB","s":"I","t":{"$date":"2021-12-26T16:22:04.43Z"},"tags":["startupWarnings"]}]})
(err = <nil>)
github.com/FerretDB/FerretDB/internal/clientconn.(*conn).run.func1
	github.com/FerretDB/FerretDB/internal/clientconn/conn.go:118
runtime.gopanic
	runtime/panic.go:844
github.com/FerretDB/FerretDB/internal/types.MustMakeDocument
	github.com/FerretDB/FerretDB/internal/types/document.go:115
github.com/FerretDB/FerretDB/internal/handlers/shared.(*Handler).MsgGetLog
	github.com/FerretDB/FerretDB/internal/handlers/shared/msg_getlog.go:76
github.com/FerretDB/FerretDB/internal/handlers.(*Handler).handleOpMsg
	github.com/FerretDB/FerretDB/internal/handlers/handler.go:170
github.com/FerretDB/FerretDB/internal/handlers.(*Handler).Handle
	github.com/FerretDB/FerretDB/internal/handlers/handler.go:85
github.com/FerretDB/FerretDB/internal/clientconn.(*conn).run
	github.com/FerretDB/FerretDB/internal/clientconn/conn.go:171
github.com/FerretDB/FerretDB/internal/clientconn.(*Listener).Run.func2
	github.com/FerretDB/FerretDB/internal/clientconn/listener.go:138
panic: types.MakeDocument: types.Document.validate: types.validateValue: unsupported type: types.Array ({[{"c":"STORAGE","ctx":"initandlisten","id":42000,"msg":"Powered by 🥭 FerretDB v0.0.4-53-g22a1644-dirty and PostgreSQL 14.1.","s":"I","t":{"$date":"2021-12-26T16:22:04.43Z"},"tags":["startupWarnings"]} {"c":"STORAGE","ctx":"initandlisten","id":42000,"msg":"Please star us on GitHub: https://github.com/FerretDB/FerretDB","s":"I","t":{"$date":"2021-12-26T16:22:04.43Z"},"tags":["startupWarnings"]}]}) [recovered]
	panic: types.MakeDocument: types.Document.validate: types.validateValue: unsupported type: types.Array ({[{"c":"STORAGE","ctx":"initandlisten","id":42000,"msg":"Powered by 🥭 FerretDB v0.0.4-53-g22a1644-dirty and PostgreSQL 14.1.","s":"I","t":{"$date":"2021-12-26T16:22:04.43Z"},"tags":["startupWarnings"]} {"c":"STORAGE","ctx":"initandlisten","id":42000,"msg":"Please star us on GitHub: https://github.com/FerretDB/FerretDB","s":"I","t":{"$date":"2021-12-26T16:22:04.43Z"},"tags":["startupWarnings"]}]})
(err = <nil>)

goroutine 40 [running]:
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00031a0c0, {0x0, 0x0, 0x0})
	go.uber.org/zap@v1.19.1/zapcore/entry.go:232 +0x60a
go.uber.org/zap.(*SugaredLogger).log(0xc00020c080, 0x3, {0xe62145, 0xd}, {0xc00027d128, 0x2, 0x2}, {0x0, 0x0, 0x0})
	go.uber.org/zap@v1.19.1/sugar.go:227 +0x13b
go.uber.org/zap.(*SugaredLogger).DPanicf(...)
	go.uber.org/zap@v1.19.1/sugar.go:154
github.com/FerretDB/FerretDB/internal/clientconn.(*conn).run.func1()
	github.com/FerretDB/FerretDB/internal/clientconn/conn.go:118 +0x1ab
panic({0xdbc700, 0xc00031c120})
	runtime/panic.go:844 +0x258
github.com/FerretDB/FerretDB/internal/types.MustMakeDocument({0xc00027d4c0, 0x6, 0x6})
	github.com/FerretDB/FerretDB/internal/types/document.go:115 +0xdc
github.com/FerretDB/FerretDB/internal/handlers/shared.(*Handler).MsgGetLog(0xc000202a38, {0xf8c480, 0xc00023e140}, 0x2?)
	github.com/FerretDB/FerretDB/internal/handlers/shared/msg_getlog.go:76 +0x1153
github.com/FerretDB/FerretDB/internal/handlers.(*Handler).handleOpMsg(0xc000220230, {0xf8c480, 0xc00023e140}, 0x13c0300?)
	github.com/FerretDB/FerretDB/internal/handlers/handler.go:170 +0x7df
github.com/FerretDB/FerretDB/internal/handlers.(*Handler).Handle(0xc000220230, {0xf8c480, 0xc00023e140}, 0xc000438c90, {0xf8cd90?, 0xc00020b920?})
	github.com/FerretDB/FerretDB/internal/handlers/handler.go:85 +0x22e
github.com/FerretDB/FerretDB/internal/clientconn.(*conn).run(0xc00023e100, {0xf8c480?, 0xc00023e140})
	github.com/FerretDB/FerretDB/internal/clientconn/conn.go:171 +0xab4
github.com/FerretDB/FerretDB/internal/clientconn.(*Listener).Run.func2()
	github.com/FerretDB/FerretDB/internal/clientconn/listener.go:138 +0x5df
created by github.com/FerretDB/FerretDB/internal/clientconn.(*Listener).Run
	github.com/FerretDB/FerretDB/internal/clientconn/listener.go:110 +0xb8e
Makefile.kev:67: recipe for target 'run' failed
make: *** [run] Error 2
```

--- a/internal/handlers/handler_test.go
+++ b/internal/handlers/handler_test.go
@@ -498,6 +498,24 @@ func TestReadOnlyHandlers(t *testing.T) {
 			),
 		},
 
+		"GetLog": {
+			req: types.MustMakeDocument(
+				"getLog", "startupWarnings",
+			),
+			resp: types.MustMakeDocument(
+				"totalLinesWritten", int32(2),
+				// will be replaced with the real value during the test
+				"log", types.MakeArray(2),
+				"ok", float64(1),
+			),
+			compareFunc: func(t testing.TB, _ types.Document, actual, expected types.CompositeType) {
+				// Just testing "ok" response, not the body of the response
+				actualV := testutil.GetByPath(t, actual, "log")
+				testutil.SetByPath(t, expected, actualV, "log")
+				assert.Equal(t, expected, actual)
+			},
+		},
+
 		"GetParameter": {
 			req: types.MustMakeDocument(
 				"getParameter", int32(1),

--- a/internal/handlers/shared/msg_getlog.go
+++ b/internal/handlers/shared/msg_getlog.go
@@ -75,7 +75,7 @@ func (h *Handler) MsgGetLog(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, 
 	err = reply.SetSections(wire.OpMsgSection{
 		Documents: []types.Document{types.MustMakeDocument(
 			"totalLinesWritten", int32(log.Len()),
-			"log", log,
+			"log", &log,
 			"ok", float64(1),
 		)},
 	})

